src/main/java/streams/PythonBridge.java

```diff
@@ -0,0 +1,146 @@
+            boolean isNameServerRunning = false;
+            for (int i = 0; i < 50; i++) {
+                String output = stdin.readLine();
+                if (output != null && output.startsWith("NS running")) {
+                    log.debug("NameServer is running");
+                    isNameServerRunning = true;
+                    break;
+                }
+                Thread.sleep(200);
+            }
+            if (!isNameServerRunning) {
+                stderr.lines().forEach(e -> log.error(e));
+                nameServerProcess.destroy();
+                throw new IOException("Timeout while waiting for start of NameServer.");
+            }
+
+
+            String[] command = {"python", "-u", pathToPythonScript}; // -u for unbuffered python output
+
+            pythonProcess = new ProcessBuilder(command).redirectErrorStream(true).start();
+            stdin = new BufferedReader(new InputStreamReader(pythonProcess.getInputStream()));
+            stderr = new BufferedReader(new InputStreamReader(pythonProcess.getErrorStream()));
+
+            boolean isPyroDaemonRunning = false;
+            for (int i = 0; i < 50; i++) {
+                String output = stdin.readLine();
+                if (output != null && output.startsWith("Pyro daemon running.")) {
+                    log.debug("Pyro daemon is running.");
+                    isPyroDaemonRunning = true;
+                    break;
+                }
+                Thread.sleep(200);
+            }
+            if (!isPyroDaemonRunning) {
+                stderr.lines().forEach(e -> log.error(e));
+                pythonProcess.destroy();
+                throw new IOException("Timeout while waiting for start of PyroDaemon.");
+            }
+
+
+            NameServerProxy nsProxy = NameServerProxy.locateNS("localhost");
+
+            remoteObject = new PyroProxy(nsProxy.lookup("streams.processors"));
+        } catch (IOException | InterruptedException e){
+            pythonProcess.destroyForcibly();
+            nameServerProcess.destroyForcibly();
+            e.printStackTrace();
+            throw new IOException("An error occured while invoking the python processes");
+        }
+
+    }
+
+    private void stop(){
+        remoteObject.close();
+        nameServerProcess.destroy();
+        pythonProcess.destroy();
+    }
+
+    /**
+     * Calls a Python method by name.
+     * This methods delegates to the {@link PyroProxy#call} method.
+     * There is some more code in here which logs python output to the appropriate java log.
+     *
+     * @param name the name of the python method to call
+     * @param args the argumeents to pass to python
+     * @return the return value of the python method. The object has to be cast back to the proper type.
+     * @throws IOException in case an error occurs when calling the python method.
+     */
+    public Object callMethod(String name, Object... args) throws IOException {
```

Comment:
this method receives DataItem, right? Shouldn't it also return such a hashmap directly? Or was your intention to enable all possible input/output combinations?